### PR TITLE
Password Reset Section

### PR DESCRIPTION
**Pull request by ahmaddanialmohd**
Included the password reset section under "Administration Issues".

### CODE REVIEW

File: source/install/troubleshooting.rst [hunk 1]
```diff
@@ -79,6 +79,24 @@ If email sign-in was turned off before the System Administrator switched sign-in
 
 4. Replace ``{username}`` with the name of the user you'd like to promote to an admin.
 
+Password Resets
+~~~~~~~~~~~~~~~
+
+Scenario #1: Internal users
```
Comment (justinegeffen) [inline source/install/troubleshooting.rst]:
```suggestion
**Email Address or Username Sign In**
```
Changed this to match the content in the provided link.

File: source/install/troubleshooting.rst [hunk 2]
```diff
@@ -79,6 +79,24 @@ If email sign-in was turned off before the System Administrator switched sign-in
...
+
+- Password reset request and emails are ONLY applicable to internal users (i.e users who are invited through invite link / email invite) based on documentation - https://docs.mattermost.com/help/getting-started/signing-in.html#email-address-or-username-sign-in.
```
Comment (justinegeffen) [inline source/install/troubleshooting.rst]:
```suggestion
- If you sign in to Mattermost using an email address or username, you can send a password reset request by selecting **I forgot my password**. 
```

File: source/install/troubleshooting.rst [hunk 3]
```diff
@@ -79,6 +79,24 @@ If email sign-in was turned off before the System Administrator switched sign-in
...
+- Since Mattermost Support do not have direct access to the server / instance the user has signed up to, we recommend users to get in touch directly to the System Administrator of the system
```
Comment (justinegeffen) [inline source/install/troubleshooting.rst]:
```suggestion
As Mattermost Support do not have direct access to your Mattermost server/instance if you experience any problems, the next step is to contact your Mattermost System Admin.
```

File: source/install/troubleshooting.rst [hunk 4]
```diff
@@ -79,6 +79,24 @@ If email sign-in was turned off before the System Administrator switched sign-in
...
+- If you are not sure who the System Administrator is, we suggest users to get in touch with the internal IT team for further help.
```
Comment (justinegeffen) [inline source/install/troubleshooting.rst]:
```suggestion
If you are not sure who the System Admin is, contact the internal IT team at your organization for further help.
```

File: source/install/troubleshooting.rst [hunk 5]
```diff
@@ -79,6 +79,24 @@ If email sign-in was turned off before the System Administrator switched sign-in
...
+
+Scenario #2 : AD / LDAP users
```
Comment (justinegeffen) [inline source/install/troubleshooting.rst]:
```suggestion
**AD/LDAP users**
```

File: source/install/troubleshooting.rst [hunk 6]
```diff
@@ -79,6 +79,24 @@ If email sign-in was turned off before the System Administrator switched sign-in
...
+
+- Password reset process should be done on the directory level instead of Mattermost UI since Mattermost pulls the information of the credentials from the LDAP directory.
```
Comment (justinegeffen) [inline source/install/troubleshooting.rst]:
```suggestion
The password reset process is performed on the directory level as Mattermost pulls the credential information from the LDAP directory.
```

File: source/install/troubleshooting.rst [hunk 7]
```diff
@@ -79,6 +79,24 @@ If email sign-in was turned off before the System Administrator switched sign-in
...
+- If the LDAP user has issues with the password, they should get in touch with the LDAP administrator for further help.
```
Comment (justinegeffen) [inline source/install/troubleshooting.rst]:
```suggestion
If you experience issues with your password for your LDAP log in, contact your organization's LDAP administrator for further help.
```

File: source/install/troubleshooting.rst [hunk 8]
```diff
@@ -79,6 +79,24 @@ If email sign-in was turned off before the System Administrator switched sign-in
...
+
+Scenario #3: SSO Users (OneLogin / Okta / etc)
```
Comment (justinegeffen) [inline source/install/troubleshooting.rst]:
```suggestion
**SSO Users (OneLogin/Okta etc)**
```
Does this also include Google and GitLab? Or just Okta and OneLogin?

File: source/install/troubleshooting.rst [hunk 9]
```diff
@@ -79,6 +79,24 @@ If email sign-in was turned off before the System Administrator switched sign-in
...
+
+- Password reset is to be done on the IdP provider side instead of Mattermost.
```
Comment (justinegeffen) [inline source/install/troubleshooting.rst]:
```suggestion
The password reset process is completed on the IdP provider side, and not via the Mattermost System Console.
```

**Comment by justinegeffen** [comment]
Hey, @ahmaddanialmohd, wanted to follow up on this so you can take a look at the suggested changes. Thank you! :)

**Comment by ahmaddanialmohd** [comment]
Looks good to me. Thank you, @justinegeffen !

File: source/install/troubleshooting.rst [hunk 10]
```diff
@@ -79,6 +79,24 @@ If email sign-in was turned off before the System Administrator switched sign-in
 
 4. Replace ``{username}`` with the name of the user you'd like to promote to an admin.
 
+Password Resets
```
Comment (justinegeffen) [inline source/install/troubleshooting.rst]:
```suggestion
Password resets
```

File: source/install/troubleshooting.rst [hunk 11]
```diff
@@ -79,6 +79,24 @@ If email sign-in was turned off before the System Administrator switched sign-in
...
+Scenario #2 : AD / LDAP users
+
+- Password reset process should be done on the directory level instead of Mattermost UI since Mattermost pulls the information of the credentials from the LDAP directory.
+- If the LDAP user has issues with the password, they should get in touch with the LDAP administrator for further help.
+
+Scenario #3: SSO Users (OneLogin / Okta / etc)
+
+The password reset process is completed on the IdP provider side, and not via the Mattermost System Console.
+
 SAML Issues
```
Comment (justinegeffen) [inline source/install/troubleshooting.rst]:
```suggestion
SAML issues
```

**Comment by justinegeffen** [comment]
> Looks good to me. Thank you, @justinegeffen !

Thanks @ahmaddanialmohd. There are a few outstanding questions - are you open to taking a look at them? If you're happy with the suggested changes you're welcome to commit them and then I'll give it another pass. :)

File: source/install/troubleshooting.rst [hunk 12]
```diff
@@ -79,7 +79,25 @@ If email sign-in was turned off before the System Administrator switched sign-in
...
+- Since Mattermost Support do not have direct access to the server / instance the user has signed up to, we recommend users to get in touch directly to the System Administrator of the system
+- If you are not sure who the System Administrator is, we suggest users to get in touch with the internal IT team for further help.
+
+**AD/LDAP users**
+
+- Password reset process should be done on the directory level instead of Mattermost UI since Mattermost pulls the information of the credentials from the LDAP directory.
+- If the LDAP user has issues with the password, they should get in touch with the LDAP administrator for further help.
+
+**SSO users (OneLogin/Okta etc)**
+
```
Comment (justinegeffen) [inline source/install/troubleshooting.rst]: Does this apply only to OneLogin and Okta?

**Comment by justinegeffen** [comment]
Hi @ahmaddanialmohd. I've committed some of my suggestions as they were stylistic. Just one question regarding the SSO content and then we're good to go. :)

File: source/install/troubleshooting.rst (hunk 12 above)
Comment (ahmaddanialmohd) [reply source/install/troubleshooting.rst]: Yes, @justinegeffen . This applies to all SSO mentioned in the [documentation](https://docs.mattermost.com/deployment/sso-saml.html). The common ones that we notice are the first two.

Comment (justinegeffen) [reply source/install/troubleshooting.rst]: Brilliant, thanks for clarifying. All looks good to me - thank you for putting this together, it's much appreciated!

**Comment by justinegeffen** [comment]
Fixes: https://github.com/mattermost/docs/pull/3077

Adding @amyblais for fresh eyes and @wiersgallak for review.

File: source/install/troubleshooting.rst [hunk 13]
```diff
@@ -79,7 +79,25 @@ If email sign-in was turned off before the System Administrator switched sign-in
...
 4. Replace ``{username}`` with the name of the user you'd like to promote to an admin.
 
-SAML Issues
+Password resets
+~~~~~~~~~~~~~~~
+
+**Email address or username sign in**
+
+If you sign in to Mattermost using an email address or username, you can send a password reset request by selecting **I forgot my password**. 
+Mattermost Support do not have access to your Mattermost server/instance. If you experience any problems, the next step is to contact your Mattermost System Admin.
```
Comment (wiersgallak) [inline source/install/troubleshooting.rst]: @justinegeffen can you confirm if this should be "Mattermost Support does not...." Not sure of the correct way as when I read it out loud, I can see both working.

Comment (justinegeffen) [inline source/install/troubleshooting.rst]:
```suggestion
The Mattermost Support team does not have access to your Mattermost server/instance. If you experience any problems, the next step is to contact your Mattermost System Admin.
```

Comment (justinegeffen) [reply source/install/troubleshooting.rst]: Good point - I've reworded it. Let me know what you think? I think it reads better.

Comment (wiersgallak) [reply source/install/troubleshooting.rst]: @justinegeffen yes, this reads better.

Comment (ahmaddanialmohd) [reply source/install/troubleshooting.rst:88]: Yes, sounds better. Thanks for the suggestion, @justinegeffen .

**Pull request merged**